modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -202,14 +204,48 @@ trait MutationMapping[F[_]: MonadCancelThrow]
           "Could not construct a subquery for the provided WHERE condition."
         )
 
-      idSelect.flatTraverse { which =>
-        observationService.use { svc =>
-          svc
-            .updateObservations(input.SET, which)
-            .map(_.map(ids => Filter(ObservationPredicates.inObservationIds(ids), child)))
+      val updateObservations: F[Result[(List[Observation.Id], Query)]] =
+        idSelect.flatTraverse { which =>
+          observationService.use { svc =>
+            svc
+              .updateObservations(input.SET, which)
+              .map(_.fproduct {
+
+                // When there are no selected ids, return a Query which matches
+                // nothing (i.e., no results for the update).
+
+                // "In" Predicate is used in the normal case where there is
+                // at least one matching observation id.  But "In" requires a
+                // non-empty list (eventually) so we need to catch the Nil
+                // case.
+
+                // Produces "Unable to map query":
+                // case Nil => Filter(False, child)
+
+                // Work around using a zero limit
+                case Nil => Limit(0, child)
+
+                case ids => Filter(ObservationPredicates.inObservationIds(ids), child)
```

Comment:
`inObservationIds(ids)` is just `In(UniquePath(List("id")), oids)`.   It allows you to construct this with an empty list, but will fail when evaluated.   It seems like `In` should work with empty lists?  If not, `Filter(False, child)` should work in that it should match nothing.

I thought of `Limit(0, child)` afterwords and maybe that's fine anyway.